### ISSUE TITLE
FATAL - Unable to connect to PostgreSQL database with postgres_session.

### ISSUE THREAD

**Issue by xanhacks** [inspec/inspec#5599]
### 🎛 Description

**Choose one:** 🐛 bug report

Hello,

If I try to login with **psql**, it works fine.

```
cloud@centos $ psql -h 127.0.0.1 -U my_username -p 5432 my_database
password for my_username: my_password

Connection successful
```

But if I try with **postgres_session**, I get an error :

```ruby
sql = postgres_session('my_username', 'my_password', '127.0.0.1', 5432)

describe sql.query('SELECT 5;', ['my_database']) do
  its('output') { should eq('5') }
end
```

```shell
cloud@centos $ inspec exec test.rb --chef-license=accept-silent
psql: FATAL:  database « my_username » does not exists
```

I set the database to "my_database" but InSpec tells me about "my_username", why ?

Regards

### 🌍 InSpec and Platform Version

CentOS 7

```shell
$ inspec --version
4.37.30
```
PostgreSQL server version 10.17
psql version 9.2.24

**Comment by Vasu1105** [comment]
@xanhacks could you please try using the latest version of the InSpec as we have fixed few issues related to postgres_conf and postgres_session resources in the new version

**Comment by boufleur** [comment]
Hi, I am having the same issue, running `inspec 4.38.9`.

**Comment by Vasu1105** [comment]
I tried to reproduce it on both 4.37.30 and 4.38.9 but was unable to reproduce it. More details @boufleur from your end will be helpful.
I tried all the below examples and those are working for me.

```
sql = postgres_session('postgres', 'mypassword')
describe sql.query('SELECT * FROM pg_shadow WHERE passwd IS NULL', ['postgres']) do
  its('output') { should eq '' }
end
```
```
sql = postgres_session('vasundhara', 'xxxxx')
describe sql.query('SELECT 5', ['postgres']) do
  its('output') { should eq "5" }
end
```
```
sql = postgres_session('vasundhara', 'xxxxxx', '127.0.0.1', 5432)
describe sql.query('SELECT 5;', ['postgres']) do
  its('output') { should eq('5') }
end
```
My version of 
psql is : psql (PostgreSQL) 12.8 and server version also the same.

**Comment by Vasu1105** [comment]
And I am executing the profile with the following command
```
bundle exec inspec exec mytest-profiles/postgre-conf-profile/ -t ssh://myvm -i ~/identityfile.pem 
```

**Comment by boufleur** [comment]
Hi @Vasu1105, I think the issue might be related with my psql version (I'm running on client 10.5, server 9.6.18).

**Comment by Vasu1105** [comment]
Thanks able to reproduce it. Fix has been given in PR #5619
